Introduction to xemc3.ipynb

```diff
@@ -0,0 +1,3171 @@
+       "<function __main__.plot_emc3grid(ip, rmin, rmax, zmin, zmax)>"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# find boundaries of the grid\n",
+    "rmin = np.min(np.asarray(R))\n",
+    "rmax = np.max(np.asarray(R))\n",
+    "zmin = np.min(np.asarray(z))\n",
+    "zmax = np.max(np.asarray(z))\n",
+    "\n",
+    "import ipywidgets as widgets\n",
+    "from ipywidgets import interact, interactive, fixed, interact_manual\n",
+    "def plot_emc3grid(ip, rmin, rmax, zmin, zmax):\n",
+    "    plt.figure(figsize=(100,100))\n",
+    "    ds.emc3.plot_Rz('Te', phi=ds.coords['phi_bounds'][ip][0], Rmin = rmin, Rmax = rmax, zmin = zmin, zmax = zmax)\n",
+    "ip = widgets.IntSlider(min = 0, max = 35, value = 0)\n",
+    "rmin_slider = widgets.FloatSlider(min = rmin, max = rmax, value = rmin, readout_format='.9f')\n",
+    "rmax_slider = widgets.FloatSlider(min = rmin, max = rmax, value =rmax, readout_format='.9f')\n",
+    "zmin_slider = widgets.FloatSlider(min = zmin, max = zmax, value = zmin, readout_format='.9f')\n",
+    "zmax_slider = widgets.FloatSlider(min = zmin, max = zmax, value = zmax, readout_format='.9f')\n",
+    "interact(plot_emc3grid, ip = ip, rmin = rmin_slider, rmax = rmax_slider, zmin = zmin_slider, zmax = zmax_slider)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "cdb7d2f0",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[91] [120] [94] [0]\n"
+     ]
+    }
+   ],
+   "source": [
+    "idx_zmin = np.where(np.isclose(-0.77468, np.asarray(z), rtol = 1e-6) == True)[0]\n",
+    "idx_zmax = np.where(np.isclose(0.82532, np.asarray(z), rtol = 5e-7) == True)[0]\n",
+    "idx_rmin = np.where(np.isclose(5.65635, np.asarray(R), rtol = 1e-9) == True)[0]\n",
+    "idx_rmax = np.where(np.isclose(5.85635, np.asarray(R), rtol = 1e-7) == True)[0]\n",
+    "\n",
+    "print(idx_zmin, idx_zmax, idx_rmin, idx_rmax)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "5f584c0d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "48ad349e74b54ecb91f2173d6d089cd5",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "interactive(children=(IntSlider(value=0, description='ip', max=35), Output()), _dom_classes=('widget-interact'…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<function __main__.plot_Te_zoomed(ip)>"
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def plot_Te_zoomed(ip):\n",
+    "    fig = plt.figure()\n",
+    "    ax = fig.add_subplot()\n",
+    "    c = plt.contourf(R[50:60, 450:460,ip],\n",
```

Comment:
Don't use cell centerd values.

Also, slicing the whole dataset, and then plotting the sliced dataset doesn't require:
1) repetition of code (copy pasting is evil)
2) error-prone indexing of plain np.arrays.

Personally I dislike countourf and would much rather see [pcolormesh](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.pcolormesh.html)